### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: '/app/saleor/static/assets'

### ISSUE THREAD

**Issue by joshuacox** [mirumee/saleor#3049]
### What I'm trying to achieve

docker-compose run web python3 manage.py collectstatic 

### Steps to reproduce the problem
1.  clone saleor
1. docker-compose build
1. docker-compose run web python3 manage.py migrate
1. then:
```
$ docker-compose run web python3 manage.py collectstatic   
Starting saleor-redis         ... done
Starting saleor-postgres      ... done
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 188, in handle
    collected = self.collect()
  File "/usr/local/lib/python3.6/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 105, in collect
    for path, storage in finder.list(self.ignore_patterns):
  File "/usr/local/lib/python3.6/site-packages/django/contrib/staticfiles/finders.py", line 125, in list
    for path in utils.get_files(storage, ignore_patterns):
  File "/usr/local/lib/python3.6/site-packages/django/contrib/staticfiles/utils.py", line 23, in get_files
    directories, files = storage.listdir(location)
  File "/usr/local/lib/python3.6/site-packages/django/core/files/storage.py", line 313, in listdir
    for entry in os.listdir(path):
FileNotFoundError: [Errno 2] No such file or directory: '/app/saleor/static/assets'
```

### What I expected to happen

not certain, but looking in that directory there is no `assets`:

```
$ docker-compose run web /bin/bash -c 'ls -alh /app/saleor/static'
Starting saleor-postgres      ... done
Starting saleor-elasticsearch ... done
total 40K
drwxr-xr-x 10 1000 1000 4.0K Oct  4 13:54 .
drwxr-xr-x 19 1000 1000 4.0K Oct  5 13:44 ..
drwxr-xr-x  5 1000 1000 4.0K Oct  4 13:54 dashboard
drwxr-xr-x 19 1000 1000 4.0K Oct  4 15:11 dashboard-next
drwxr-xr-x  2 1000 1000 4.0K Oct  4 13:54 favicons
drwxr-xr-x  2 1000 1000 4.0K Oct  4 13:54 fonts
drwxr-xr-x  2 1000 1000 4.0K Oct  4 13:54 images
drwxr-xr-x  4 1000 1000 4.0K Oct  4 13:54 js
drwxr-xr-x  8 1000 1000 4.0K Oct  4 13:54 placeholders
drwxr-xr-x  4 1000 1000 4.0K Oct  4 13:54 scss
```

**System information**
Operating system:  Docker on Arch Linux
Browser: brave,opera,vivaldi,konqueror,firefox,chromium

commit:
```
commit 0b5c92d00c0cd76f73b98d278ac12c0fee824847
Merge: ff54fe2e7 f8d0b2ac6
Author: Marcin Gębala <5421321+maarcingebala@users.noreply.github.com>
Date:   Fri Oct 5 15:28:41 2018 +0200

    Merge pull request #3040 from jxltom/fixes-deprecated-warning-for-latest-pytest
    
    Fixed tons of deprecated warnings using latest pytest
```

**Comment by patrys** [comment]
I think that's caused by the current working dir being mounted at `/app` inside the container by docker-compose. It's done for convenience so that modifying code won't require the image to be rebuilt but it ends up shadowing the static file directory that is built as part of image construction.

**Comment by joshuacox** [comment]
@patrys  do you know the preferred method for making that directory?

This is the incorrect method:

```
mkdir saleor/static/assets
chmod 777 saleor/static/assets
ln -s saleor/static ./
chmod 777 saleor/static
mkdir static/js/payments
chmod 777 static/js/payments
```

as this seems hackish, but those are the steps I got it working with.

Are these directories not created properly by a script somewhere?

Perhaps a step is missing here:
https://saleor.readthedocs.io/en/latest/customization/docker.html

**Comment by maarcingebala** [comment]
Should be resolved in #3091.
